src/mds/PurgeQueue.cc

```diff
@@ -348,6 +348,7 @@ bool PurgeQueue::_consume()
 {
   assert(lock.is_locked_by_me());
 
+  uint64_t old_read_pos = journaler.get_read_pos();
   bool could_consume = false;
   while(can_consume()) {
     could_consume = true;
```

Comment:
no need to introducing old_read_pos, just move 'could_consume = true' to the place after journaler.is_readable() check